pelican/readers.py

```diff
@@ -246,8 +246,13 @@ class MarkdownReader(BaseReader):
 
     def __init__(self, *args, **kwargs):
         super(MarkdownReader, self).__init__(*args, **kwargs)
-        self.extensions = self.settings['MD_EXTENSIONS']
-        self.extensions.setdefault('markdown.extensions.meta', {})
+        # make sure 'extension_configs' exists and
+        # and either way 'markdown.extensions.meta' must be in there
+        settings = self.settings['MARKDOWN_SETTINGS']
+        settings.setdefault('extension_configs', {})
+        settings['extension_configs'].setdefault(
+            'markdown.extensions.meta', {})
+        settings['extensions'] = list(settings['extension_configs'].keys())
```

Comment:
I found this line confusing.  Since we're passing kwargs to markdown, I was expecting `MARKDOWN_SETTINGS = { 'extensions': [ 'my_custom_extension' ] }` to work in the config.  Perhaps it would be better to append?

```
settings.setdefault('extension', [])
settings['extensions'].append(list(settings['extension_configs'].keys()))
```

..or the like.